src/muse/utilities.py

```diff
@@ -238,47 +243,41 @@ def broadcast_techs(
         We want to select the values from the technology array that correspond to each
         asset in the template. To do this, we perform `broadcast_techs` on
         `technologies` using `assets` as a template:
-        >>> broadcast_techs(technologies, assets)
+        >>> broadcast_techs(technologies, assets, installed_as_year=False)
         <xarray.DataArray (asset: 2)> Size: 16B
         array([1, 5])
         Coordinates:
             technology  (asset) <U9 72B 'gasboiler' 'heatpump'
             region      (asset) <U2 16B 'R1' 'R2'
         Dimensions without coordinates: asset
 
-        The output array has the same shape as the assets template. Each value in the
-        output is the value in the original technology array that matches the
+        The output array has an "asset" dimension which matches the template. Each value
+        in the output is the value in the original technology array that matches the
         technology & region of each asset.
     """
-    # this assert will trigger if 'year' is changed to 'installed' in
-    # technologies, because then this function should be modified.
-    assert "installed" not in technologies.dims
+    assert "asset" in template.dims
+
+    # Name of asset coordinates (e.g. "technology", "region", "installed")
     names = [u for u in template.coords if template[u].dims == ("asset",)]
-    # the first selection reduces the size of technologies without affecting the
+    assert "year" not in names
+
+    # If installed_as_year is True, we need to rename the installed dimension to "year"
+    # TODO: this should be stricter, and enforce that the template has "installed" data,
+    # and that the technologies dataset has a "year" dimension.
+    # if installed_as_year:
```

Comment:
As there seem to be more things to do, maybe implement the changes I suggest above as a separate PR, alongside this TODO.